CIP-XXXX/README.md

```diff
@@ -0,0 +1,504 @@
+      "type": "number",
+      "title": "APIErrorCode"
+      "enum": [-1, -2, -3, -4, -5]
+    },
+    "info": {
+      "type": "string"
+    }
+  },
+  "required": ["code"],
+}
+```
+
+- `InvalidRequest`: (-1) Inputs do not conform to this spec or are otherwise invalid.
+- `InternalError`: (-2) An error occurred during execution of this API call.
+- `Refused`: (-3) The request was refused due to lack of access - e.g. wallet disconnects.
+- `AccountChange`: (-4) The account has changed. The dApp should call wallet.enable() to reestablish connection to the new account. The wallet should not ask for confirmation as the user was the one who initiated the account change in the first place.
+- `NotSatisfiable`: (-5) The request is structurally correct, but the wallet can not satisfy it for some reason.
+
+##### DataSignError
+
+```
+{
+  "type": "object",
+  "title": "DataSignError",
+  "properties": {
+    "code": {
+      "type": "number",
+      "title": "DataSignErrorCode"
+      "enum": [1, 2, 3]
+    },
+    "info": {
+      "type": "string"
+    }
+  },
+  "required": ["code"],
+}
+```
+
+- `ProofGeneration`: (1) Wallet could not sign the data (e.g. does not have the secret key associated with the address).
+- `AddressNotPK`: (2) Address was not a P2PK address and thus had no SK associated with it.
+- `UserDeclined`: (3) User declined to sign the data.
+
+##### TxSendError
+
+```
+{
+  "type": "object",
+  "title": "TxSendError",
+  "properties": {
+    "code": {
+      "type": "number",
+      "title": "TxSendErrorCode"
+      "enum": [1, 2]
+    },
+    "info": {
+      "type": "string"
+    }
+  },
+  "required": ["code"],
+}
+```
+
+- `Refused`: (1) Wallet refuses to send the tx (could be rate limiting).
+- `Failure`: (2) Wallet could not send the tx.
+
+##### TxSignError
+
+```
+{
+  "type": "object",
+  "title": "TxSignError",
+  "properties": {
+    "code": {
+      "type": "number",
+      "title": "TxSignErrorCode"
+      "enum": [1, 2]
+    },
+    "info": {
+      "type": "string"
+    }
+  },
+  "required": ["code"],
+}
+```
+
+- `ProofGeneration`: (1) User has accepted the transaction sign, but the wallet was unable to sign the transaction (e.g. not having some of the private keys).
+- `UserDeclined`: (2) User declined to sign the transaction.
+
+## Rationale: how does this CIP achieve its goals?
+
+The goal of this CIP is to define an extension to enable an "own-data wallet". We build upon the work done in [CIP-0144 | Wallet Connector API](link) and define an extension compatible with that CIP.
+
+## Path to Active
+
+CIP-0144 includes a [section](link/#CIP-30-Backwards-compatibility) which introduces the idea of a wrapper, that takes a CIP-30 compatible API, and makes it CIP-0144 compatible. Since in this CIP we are simply replicating the functionality offered by CIP-30, something very similar would be possible here. Implementors of this wrapper must take care of the following differences between this CIP and CIP-30.
```

Comment:
```suggestion
CIP-0144 includes a [section](https://github.com/nazrhom/CIPs/blob/nazrhom/full-data-wallet-connector/CIP-0144/README.md#cip-30-backwards-compatibility) which introduces the idea of a wrapper, that takes a CIP-30 compatible API, and makes it CIP-0144 compatible. Since in this CIP we are simply replicating the functionality offered by CIP-30, something very similar would be possible here. Implementors of this wrapper must take care of the following differences between this CIP and CIP-30.
```